### ISSUE TITLE
catkin/groovy support?

### ISSUE THREAD

**Issue by rctoris** [youbot/youbot-ros-pkg#21]
Any word on if Groovy has been looked at with the youBot or if consideration has been made to look into migrating to catkin?

Also, might it be better to break up all of the various stacks (metapackages) into separate repositories and adhere closer to the branching strategies in ROS (e.g., groovy-devel, hydro-devel...). This would also make it a lot easier to eventually add support to integrate into the ROS build system to create debian packages for the youBot.

Thoughts? :hamster:

**Comment by wnowak** [comment]
You can find some provisional catkinized versions for groovy on my account:
https://github.com/wnowak/youbot_driver
https://github.com/wnowak/youbot-ros-pkg
https://github.com/wnowak/brics_actuator
https://github.com/wnowak/youbot_applications

In the youbot-ros-pkg only the ROS wrapper (i.e. youbot_oodl package) is catkinized. There are still some things and tests missing. I used some workaround to get the dependencies on brics_actuator and pr2_msgs in catkin which are not so nice.

I think it could make much sense to break up the stacks as you suggest, we will look into that.

**Comment by rctoris** [comment]
Great, thanks for the update @wnowak . We stumbled upon your catkinized version of youbot_driver yesterday and already merged it into our fork. Our group currently is working on getting youbot_driver into a stable state that can be built on the ROS build farm and distributed as packages for Ubuntu. The work you have done with catkin has helped a great deal. 

One change we did make to the basic code was to read the folder location of the config files from a environment variable rather than a relative path (e.g., https://github.com/youbot/youbot_driver/blob/master/youbot/YouBotBase.hpp#L81). We also added the `testing` folder back into the build script. For next steps, we wanted to clean-up the package structure a bit possibly by creating more-standard `src` and `include` folders. 

I noticed that some of the CMake files are configured to allow both a catkin and non-catkin build. Since catkin really is just a build system that can be installed independent of ROS, can we not just support the catkin conveniences? Just a thought.

Let us know if any of these suggestions are worth pulling in as we'd love to contribute back and get involved with the youBot community. 

@adamjardim @andywolff
